### ISSUE TITLE
CUDA error 9: invalid configuration argument

### ISSUE THREAD

**Issue by Gigioliva** [catboost/catboost#2228]
Problem:

When I run a train using CatBoostRegressor using **4** GPUs I get a configuration error:
```
Application terminated with error: ??+0 (0x7F5490A7AB94)
??+0 (0x7F5491E7C73B)
??+0 (0x7F5491E7B7AD)
??+0 (0x7F5491E7C968)
??+0 (0x7F5490D022F3)
??+0 (0x7F5490D0215A)
??+0 (0x7F55434D6B43)
clone+68 (0x7F5543567BB4)
(TCatBoostException) catboost/cuda/cuda_lib/cuda_base.h:261: CUDA error 9: invalid configuration argument
```

Running the same train with **2** GPUs instead everything works correctly.

catboost version: 1.1.1
Operating System: ubuntu 22.04
CPU: Intel Haswell x86/64
GPU: 4 x NVIDIA T4

**Comment by Gigioliva** [comment]
UPDATE: by downgrading the catboost version to `1.1.0`, the problem does not appear

**Comment by ek-ak** [comment]
Hello! Thanks for you report!
Could you provide a reproducer if it is possible ?

**Comment by ZaydH** [comment]
I ran into an issue very similar to @Gigioliva .  However, my error appears in `cuda_util/sort.cpp` while @Gigioliva 's issue was in `cuda_lib/cuda_base.h`.  

My error message:

```
Application terminated with error: ??+0 (0x7F9BB67CF99F)
??+0 (0x7F9BB7A592B4)
??+0 (0x7F9BB7A5A968)
??+0 (0x7F9BB68CB2F3)
??+0 (0x7F9BB68CB15A)
??+0 (0x7F9CB8494B43)
??+0 (0x7F9CB8526A00)

(TCatBoostException) catboost/cuda/cuda_util/sort.cpp:166: CUDA error 9: invalid configuration argument
```

My configuration:
* CatBoost Version: 1.1.1
* Operating System: PopOS 22.04
* CPU: AMD Ryzen 9 5950X 
* GPU: 4090
* CUDA: 11.8

I also confirmed that downgrading to CatBoost 1.1.0 immediately fixed the problem.

I cannot share my code publicly but may be able to share it privately if there is still interest.
